File: datastructures/symboltables/SequentialSearchST.go
package main

import "fmt"

type SequentialSearchST struct {
    first *Node;
}

type Node struct {
    key string;
    val int;
    next *Node;
}

func (st *SequentialSearchST) get(key string) int {
    for x := st.first; x != nil; x = x.next {
        if key == x.key {
            return x.val
        }
    }
    return -1
}

func (st *SequentialSearchST) put(key string, val int) {
    for x := st.first; x != nil; x = x.next {
        if key == x.key {
            x.val = val
            return
        }
    }
    st.first = &Node{key, val, st.first}
}

func main() {
    st := new(SequentialSearchST)

    st.put("a", 0)
    st.put("b", 1)
    st.put("c", 2)

    fmt.Println(fmt.Sprintf("c: %d", st.get("c")))
    fmt.Println(fmt.Sprintf("b: %d", st.get("b")))
    fmt.Println(fmt.Sprintf("a: %d", st.get("a")))
}

File: datastructures/symboltables/SequentialSearchST.cpp
#include <iostream>
#include <vector>

using namespace std;


struct Node;

struct Node {
    string key;
    int val;
    Node *next;
};


class SequentialSearchST {

private:
    Node *first;

public:
    SequentialSearchST();
    int get(string);
    void put(string, int);
};

int SequentialSearchST::get(string key) {
    for (Node *x = first; x != nullptr; x = x->next) {
        if (x->key == key) {
            return x->val;
        }
    }

    return -1;
}

void SequentialSearchST::put(string key, int val) {
    for (Node *x = first; x != nullptr; x = x->next) {
        if (x->key == key ) {
            x->val = val;
            return;
        }
    }
    // TODO -- I suspect there is a more compact way to do this, need to figure it out.
    Node *temp = new Node();
    temp->key = key;
    temp->val = val;
    temp->next = first;
    first = temp;
}

SequentialSearchST::SequentialSearchST() {
    return;
}


int main() {

     SequentialSearchST *st = new SequentialSearchST();

    st->put("a", 2);
    st->put("b", 3);
    st->put("c", 4);

    cout << "c: " << st->get("c") << endl;
    cout << "b: " << st->get("b") << endl;
    cout << "a: " << st->get("a") << endl;

    return 0;
}
